src/psim/fc/orbit_controller.cpp

```diff
@@ -45,22 +45,51 @@ void OrbitController::step() {
 
   auto const &t = truth_t_s->get();
   auto const &t_ns = truth_t_ns->get();
-  auto const &r_ecef = truth_satellite_orbit_r_ecef->get();
-  auto const &v_ecef = truth_satellite_orbit_v_ecef->get();
-  auto const &other_r_ecef = truth_other_orbit_r_ecef->get();
-  auto const &other_v_ecef = truth_other_orbit_v_ecef->get();
+
   auto &J_ecef = truth_satellite_orbit_J_ecef->get();
   auto const &m = truth_satellite_m.get();
   auto &cumulative_dv = fc_satellite_cumulative_dv.get();
+  auto const &relative_orbit_is_valid = fc_satellite_relative_orbit_is_valid->get();
+  auto const &r_ecef = fc_satellite_orbit_r->get();
+  auto const &v_ecef = fc_satellite_orbit_v->get();
+  auto const &other_r_ecef = truth_other_orbit_r_ecef->get();
+  auto const &other_v_ecef = truth_other_orbit_v_ecef->get();
+  auto const &fire_time = fc_satellite_fire_time.get();
+
+  Vector3 dr_ecef;
+  Vector3 dv_ecef;
+  Vector3 dr_ecef_smoothed;
+  Vector3 dv_ecef_smoothed;
+
+
+  if (relative_orbit_is_valid) {
+    dr_ecef = fc_satellite_relative_orbit_dr->get();
+    dv_ecef = fc_satellite_relative_orbit_dv->get();
+  } else {
+    dr_ecef = (other_r_ecef - r_ecef);
+    dv_ecef = (other_v_ecef - v_ecef);
+  }
+
+  if(!lin::all(lin::isfinite(prev_dr_ecef))){
+    dr_ecef_smoothed=dr_ecef;
+  } else {
+    dr_ecef_smoothed=alpha*dr_ecef+(1.0-alpha)*prev_dr_ecef;
+  }
+
+  if(!lin::all(lin::isfinite(prev_dv_ecef))){
+    dv_ecef_smoothed=dv_ecef;
+  } else {
+    dv_ecef_smoothed=alpha*dv_ecef+(1.0-alpha)*prev_dv_ecef;
+  }
```

Comment:
I don't think we're ever actually doing any smoothing here? Like we never assign back to `prev_d*_ecef` so it's the same as reading data straight from the estimator.